NutritionManagement/src/main/Day.java

```diff
@@ -20,11 +21,8 @@ public Day(String dateString) {
 	private ArrayList<FoodPortion> foodPortions;
 
 	public NutritionValue getTotalNutrition() {
-		Stream<NutritionValue> nutritions = foodPortions.stream().map(foodPortion -> foodPortion.getNutrition());
-		return new NutritionValue(
-				nutritions.map(item -> item.getCalories()).reduce(0.0, (prev, curr) -> prev + curr),
-				nutritions.map(item -> item.getProtein()).reduce(0.0, (prev, curr) -> prev + curr),
-				nutritions.map(item -> item.getCarbs()).reduce(0.0, (prev, curr) -> prev + curr),
-				nutritions.map(item -> item.getFat()).reduce(0.0, (prev, curr) -> prev + curr));
+		ArrayList<NutritionValue> nutritions = foodPortions.stream().map(foodPortion -> foodPortion.getNutrition())
```

Comment:
Consider checking if foodPortions is null or not empty before invoking **’NutritionCalculator.accumulateNutritionValues(foodPortions.stream()...)‘**
 to prevent potential **“NullPointerException“**.

Reply:
I'm still studying Java Stream (previously I use TS and apparently Java is different). I'll update it when necessary.